tree-sitter/src/TreeSitter/Unmarshal.hs

```diff
@@ -258,17 +260,15 @@ instance GUnmarshal U1 where
 
 -- For unary products:
 instance (Selector s, Unmarshal k) => GUnmarshal (M1 S s (K1 c k)) where
-  gunmarshalNode _ = push $ do
-    fields <- getFields
-    gunmarshalProductNode fields
+  gunmarshalNode _ = push getFields >>= gunmarshalProductNode . fromMaybe Map.empty
```

Comment:
1. I rewrote this to mirror the flow in the instance for `f :*: g`, below.
2. I made the `push` only occur around the call to `getFields`. This behaviour is actually exactly equivalent for the extant uses of this instance, since they’re all anonymous nodes anyway, and thus have no children, and thus the `push` behaviour would have left the cursor pointing at the parent node either way.
3. Then, to accommodate the `Maybe`, I added the `fromMaybe` bit.